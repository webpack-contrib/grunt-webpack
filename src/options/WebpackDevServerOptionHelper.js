"use strict";

const defaults = require("./default");
const OptionHelper = require("./OptionHelper");

class WebpackOptionHelper extends OptionHelper {
  getDefaultOptions() {
    return Object.assign(
      {},
      defaults.gruntOptions,
      defaults.webpackDevServerOptions,
    );
  }

  getWebpackOptions() {
    const options = this.getOptions();

    if (Array.isArray(options)) {
      return options.map((opt) => this.filterGruntOptions(opt));
    }

    return this.filterGruntOptions(options);
  }

  getWebpackDevServerOptions() {
    const options = this.getOptions();

    if (Array.isArray(options)) {
      return options.reduce(
        (previous, current) => ({
          ...previous,
          ...current.devServer,
        }),
        {},
      );
    }

    return options.devServer;
  }
}

module.exports = WebpackOptionHelper;
