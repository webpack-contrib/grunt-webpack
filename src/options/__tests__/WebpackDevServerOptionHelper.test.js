import WebpackDevServerOptionHelper from "../WebpackDevServerOptionHelper";

describe("WebpackDevServerOptionHelper", () => {
  test("host is localhost by default", () => {
    const options = {};
    const helper = new WebpackDevServerOptionHelper();

    helper.readRawConfig = () => options;

    helper.preloadOptions(() => {
      expect(helper.getWebpackOptions()).toMatchSnapshot();
      expect(helper.getWebpackDevServerOptions()).toMatchSnapshot();
    });
  });

  test("supports array config for webpack", () => {
    const options = [
      { devServer: { port: 8080 } },
      { devServer: { port: 9090 } },
    ];
    const helper = new WebpackDevServerOptionHelper();

    helper.getOptions = () => options;

    expect(helper.getWebpackOptions()).toMatchSnapshot();
    expect(helper.getWebpackDevServerOptions()).toMatchSnapshot();
  });
});
