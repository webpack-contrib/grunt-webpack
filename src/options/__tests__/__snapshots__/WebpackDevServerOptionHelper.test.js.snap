// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WebpackDevServerOptionHelper host is localhost by default 1`] = `
Object {
  "devServer": Object {
    "host": "localhost",
  },
  "stats": Object {
    "cached": false,
    "cachedAssets": false,
    "colors": true,
  },
}
`;

exports[`WebpackDevServerOptionHelper host is localhost by default 2`] = `
Object {
  "host": "localhost",
}
`;

exports[`WebpackDevServerOptionHelper supports array config for webpack 1`] = `
Array [
  Object {
    "devServer": Object {
      "port": 8080,
    },
  },
  Object {
    "devServer": Object {
      "port": 9090,
    },
  },
]
`;

exports[`WebpackDevServerOptionHelper supports array config for webpack 2`] = `
Object {
  "port": 9090,
}
`;
